File: src/main/scala/scalaapi/AlertingService.scala
package io.lenses.alerting.plugin.scalaapi

import io.lenses.alerting.plugin.{Alert, AlertingService => JAlertingService}

import scala.util.Try

/**
  * simple alerting service wrapper giving Scala users a bit more type information
  */
class AlertingService(service: JAlertingService) {

  def publish(alert: Alert): Try[Unit] = Try { service.publish(alert) }
}

File: src/main/java/io/lenses/alerting/plugin/AlertingService.java
package io.lenses.alerting.plugin;

import io.lenses.alerting.plugin.util.Try;

public interface AlertingService {
    /**
     * Publishes event to alerting service.
     * @param alert alert to be published
     * @return nothing if successful, failure if not
     */
    Try<Void> publish(Alert alert);
}
